src/controlplane/controlplane_test.go

```diff
@@ -101,8 +101,16 @@ func Test_Scraper_Autodiscover_all_cp_components(t *testing.T) {
 				t.Fatalf("running scraper: %v", err)
 			}
 
+			// Include specific specific exclusions that depend on version.
+			versionAsserter := asserter
+			// apiserverStorageObjects replaces etcObjectCounts in k8s versions above 1.23
+			if testutil.IsBelow(version, testutil.Testdata123) {
+				versionAsserter = versionAsserter.Excluding(exclude.Metrics("apiserverStorageObjects"))
+			} else {
+				versionAsserter = versionAsserter.Excluding(exclude.Metrics("etcdObjectCounts"))
+			}
```

Comment:
Minor stuff, but I wonder why you chose to use a different variable in this case?
```suggestion
			// apiserverStorageObjects replaces etcObjectCounts in k8s versions above 1.23
			if testutil.IsBelow(version, testutil.Testdata123) {
				asserter = versionAsserter.Excluding(exclude.Metrics("apiserverStorageObjects"))
			} else {
				asserter = versionAsserter.Excluding(exclude.Metrics("etcdObjectCounts"))
			}
```

Reply:
I'm using a different variable because `asserter` is defined outside the loop. `t.Parallel` is being used so the behaviour if we keep the same variable becomes unpredictable, besides if subtests weren't running in parallel, the last iteration would include the exclusions from all the previous iterations. Maybe keeping the same name could be more idiomatic (`asserter := asserter`) but I think a different one is more readable.

Reply:
Fair enough!